### PR TITLE
Add a working WeakRealmNotifier for Node

### PR DESCRIPTION
**Pull request by appden**
Uses libuv APIs to async schedule onto the default Node run loop.

### CODE REVIEW

File: src/object-store/src/impl/node/weak_realm_notifier.cpp [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    rgt.m_handle = nullptr;
+
+    return *this;
+}
+
+WeakRealmNotifier::~WeakRealmNotifier()
+{
+    if (m_handle) {
+        uv_close((uv_handle_t*)m_handle, [](uv_handle_t* handle) {
+            auto realm_weak_ptr = static_cast<std::weak_ptr<Realm>*>(handle->data);
```
Comment (alazier) [inline src/object-store/src/impl/node/weak_realm_notifier.cpp]: are these three calls safe/correct given that they are made async?

Comment (appden) [inline src/object-store/src/impl/node/weak_realm_notifier.cpp]: Yes, the handle is created in one place and destroyed in one place. There's no concurrent access to it.

**Comment by alazier** [comment]
What tests that this actually works properly?

**Comment by appden** [comment]
The fact it compiles and doesn't crash. :smile: I have a branch where this is tested, but relies on merging other PRs, so I'm working toward getting that into master.

**Comment by appden** [comment]
@alazier I would really like to merge this

**Comment by alazier** [comment]
We need to have automated tests. What other prs do the tests rely on?

**Comment by appden** [comment]
Once I incorporate async tests, then this will be tested. This PR does not affect a released product yet.

**Comment by alazier** [comment]
Understood - but when I make changes I want to be able to verify that I am not breaking this functionality.

**Comment by appden** [comment]
Async tests are coming in another PR very soon. I don't see how you'll break this in the meantime. :smile:

**Pull request merged**